actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -191,41 +198,47 @@ pub fn call<RT: Runtime>(
                     )
                 }
 
-                CallKind::CallCode => {
-                    todo!()
-                }
+                CallKind::CallCode => Err(ActorError::unchecked(
+                    EVM_CONTRACT_EXECUTION_ERROR,
+                    "unsupported opcode".to_string(),
+                )),
             };
             match call_result {
                 Ok(result) => {
                     // Support the "empty" result. We often use this to mean "returned nothing" and
                     // it's important to support, e.g., sending to accounts.
                     if result.is_empty() {
-                        Vec::new()
+                        (1, Vec::new())
                     } else {
                         // TODO: support IPLD codecs #758
                         let BytesDe(result) = result.deserialize()?;
-                        result
+                        (1, result)
                     }
                 }
-                Err(ae) => {
-                    return if ae.exit_code() == EVM_CONTRACT_REVERTED {
-                        // reverted -- we don't have return data yet
-                        // push failure
-                        stack.push(U256::zero());
-                        Ok(())
-                    } else {
-                        Err(StatusCode::from(ae))
-                    };
-                }
+                Err(ae) => match ae.exit_code() {
+                    EVM_CONTRACT_REVERTED | EVM_CONTRACT_EXECUTION_ERROR => (0, ae.data().to_vec()),
+                    _ => {
+                        let msg_data = {
+                            let data = ae.data();
+                            if data.is_empty() {
+                                ae.msg().as_bytes().to_vec()
```

Comment:
Hm... this makes me uncomfortable. I know it helps with debugging... but it still really makes me uncomfortable.

Can we just rely on tracing/backtraces?

Reply:
I guess there will be some form of backtrace here, so ok , we can remove this. We can always readd if turns out its hard to debug without it.